core/services/registrysyncer/syncer.go

```diff
@@ -211,20 +264,50 @@ func (s *registrySyncer) sync(ctx context.Context) error {
 		s.reader = reader
 	}
 
-	state, err := s.state(ctx)
+	lr, err := s.localRegistry(ctx)
 	if err != nil {
 		return fmt.Errorf("failed to sync with remote registry: %w", err)
 	}
 
 	for _, h := range s.launchers {
-		if err := h.Launch(ctx, state); err != nil {
+		if err := h.Launch(ctx, lr); err != nil {
 			s.lggr.Errorf("error calling launcher: %s", err)
 		}
 	}
 
 	return nil
 }
 
+func toCapabilityType(capabilityType uint8) capabilities.CapabilityType {
+	switch capabilityType {
+	case 0:
+		return capabilities.CapabilityTypeTrigger
+	case 1:
+		return capabilities.CapabilityTypeAction
+	case 2:
+		return capabilities.CapabilityTypeConsensus
+	case 3:
+		return capabilities.CapabilityTypeTarget
+	default:
+		// Not found
+		return capabilities.CapabilityType(-1)
```

Comment:
this is alarming. it's overflow. i don't understand why it compiles tbh. isn't CapabilityType a uint8?
https://go.dev/play/p/1lcAh4kPNE2

typical the default value is used for UNKNOWN. is it too late to change that? if so, let's define the max value as UNKNOWN in the lib and return that here.

Reply:
CapabilityType is an int, which is why this isn't erroring